sdks/python/apache_beam/runners/direct/executor.py

```diff
@@ -271,6 +272,8 @@ def __init__(self, transform_evaluator_registry, evaluation_context,
     self._side_input_values = {}
     self.blocked = False
     self._call_count = 0
+    self._retry_count = 0
+    self._MAX_RETRY_PER_BUNDLE = 4
```

Comment:
<!--new_thread; commit:eb2acfa855db99ab4b826344656ec29dcdb175e3; resolved:0-->
You can factor out this to be a constant outside TransformExecutor.